js/ui/file_manager/ui.file_manager.toolbar.js

```diff
@@ -197,20 +157,51 @@ class FileManagerToolbar extends Widget {
         });
     }
 
-    _getItemConfigByCommandName(commandName) {
+    _configureItemByCommandName(commandName, item) {
+        let result = {};
+
         const command = this._commandManager.getCommandByName(commandName);
         if(command) {
-            return this._createCommandItem(command);
+            result = this._createCommandItem(command);
         }
 
         switch(commandName) {
             case "separator":
-                return this._createSeparatorItem();
+                result = this._createSeparatorItem();
+                break;
             case "viewMode":
-                return this._createViewModeItem();
+                result = this._createViewModeItem();
+                break;
         }
 
-        return {};
+        if(this._isDefaultItem(commandName)) {
+            const defaultConfig = DEFAULT_ITEM_CONFIGS[commandName];
+            extend(result, defaultConfig);
+            extend(result, item.visibilityMode ? { visibilityMode: item.visibilityMode } : {});
+            if(result.visibilityMode) {
```

Comment:
I think we should check `result.visibilityMode === "manual"` because `result.visibilityMode` can be explicit specified as `"auto"`.